src/url-replacements.js

```diff
@@ -94,6 +98,47 @@ class UrlReplacements {
             Promise.resolve());
       });
     });
+
+    // Returns the number of milliseconds since 1 Jan 1970 00:00:00 UTC.
+    this.set_('TIMESTAMP', () => {
+      return new Date().getTime();
+    });
+
+    // Returns the user's time-zone offset from UTC, in minutes.
+    this.set_('TIMEZONE', () => {
+      return new Date().getTimezoneOffset();
+    });
+
+    // Returns a promise resolving to viewport.getScrollTop.
+    this.set_('SCROLL_TOP', () => {
+      return new Promise(resolve => {
+        resolve(viewportFor(this.win_).getScrollTop());
```

Comment:
This promise doesn't help unfortunately. The main part here is use of vsync. I posted the fragment of the code necessary.